### ISSUE TITLE
how save the value of a tensorflow operation in an array

### ISSUE THREAD

**Issue by molo32** [tensorflow/tensorflow#44186]
save the value of a tensorflow operation in an array.

i am using tensorflow v2 in jupyter

I have a code that calculates the loss error, I would like to save the value for each time in the gen_loss_epoch array. I try to save the error value with the following code

`gen_loss_epoch[epoch] = tf.print (gen_loss)`

while training the error value is printed eg 0.990000 0.800 0.2000 etc thanks to tf.print (gen_loss)

I thought that if I did ggg [time] = tf.print (gen_loss) it would save those loss values in the gen_loss_epoch array.

```
gen_loss_epoch=np.linspace(0,0,100)

@tf.function()
def train_step(input_image, target,epoch):
    global gen_loss_epoch
    
    with tf.GradientTape() as gen_tape, tf.GradientTape() as discr_tape:
        
        output_image = generator(input_image, training=True)

        output_gen_discr = discriminator([output_image, input_image], training=True)

        output_trg_discr = discriminator([target, input_image], training=True)

        discr_loss = discriminator_loss(output_trg_discr, output_gen_discr)
      
        gen_loss = generator_loss(output_gen_discr, output_image, target)
        
        #with tf.Session() as sess:
        tf.print("error del discriminador=", discr_loss)
        tf.print("error del generador=",  gen_loss)

        gen_loss_epoch[epoch]=tf.print(gen_loss)
        generator_grads = gen_tape.gradient(gen_loss, generator.trainable_variables)
        
        discriminator_grads = discr_tape.gradient(discr_loss, discriminator.trainable_variables)
        
        generator_optimizer.apply_gradients(zip(generator_grads, generator.trainable_variables))
        
        discriminator_optimizer.apply_gradients(zip(discriminator_grads, discriminator.trainable_variables))
```
        
but when he finishes training I see that

  >> type(gen_loss_epoch) 
  <tf.Operation 'PrintV2_2' type = PrintV2>
they are not numbers

Why does it return a tf.Operation? Why doesn't it save me the error value as it shows when training and just save a <tf.Operation 'PrintV2_2' type = PrintV2>?

**Comment by amahendrakar** [comment]
@molo32,
With `gen_loss_epoch[epoch] = tf.print (gen_loss)`, you are assigning the `tf.print` method to the array instead of the value of `gen_loss`. Instead please try 

 - `gen_loss_epoch[epoch] = gen_loss`, if `gen_loss` is a number or
 - `gen_loss_epoch[epoch] = gen_loss.numpy()`, if `gen_loss` is a Tensor.

Also, this question is better asked on [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow) since it is not a TensorFlow bug or feature request. There is also a larger community that reads questions there. Thanks!
